### PR TITLE
Finding last digit of nth term in squares of fibonacci number

### LINKED ISSUE #1696: Finding last digit of nth term in squares of fibonacci number by c++
I will give optimised code for finding last digit square of nth fibonaaci number

### PR DESCRIPTION
**Pull request by RedEye00**
Fixes #1696 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request: 
- keep space before and after operators
-keep a space after ; in for loop
-include some comments to explain the operation
-please keep space after comma
-add two sample input and output
-please include this file inside the appropriate folder, if the folder is not exist create a new one
-please correct the file name
-please include meaningful commit message

#### Languages Used:
- C++
#### Files Added:
- Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp

Thanks!

### CODE REVIEW

File: Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//C++ Program to find last digit of square of nth fibonacci numbers
+#include<iostream>
+using namespace std;
+
+int main(){
+    int a[60], n, i, c, b[60], j;
+    a[0] = 0;
+    a[1] = 1;
+    b[0] = 0;
+    b[1] = 1;
```
Comment (asha15) [inline Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp]: keep a blank line after this

File: Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    //Using the concept that after 60 numbers in a fibonacci numbers the last digit repeats itself again and again
```
Comment (asha15) [inline Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp]: make this into multi line comment

File: Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    for(i = 2 ; i < 60 ; i++ ){        //for storing the last digits of squares of first 60 fibonacci numbers
```
Comment (asha15) [inline Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp]: remove the space after i++
can i know why did you include two comment into the same function? it's better if you declare a one comment

File: Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        a[i] = (a[i-1] + a[i-2]) % 10;
```
Comment (asha15) [inline Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp]: keep space around -

File: Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    int a[60], n, i, c, b[60], j;
+    a[0] = 0;
+    a[1] = 1;
+    b[0] = 0;
+    b[1] = 1;
+    //Using the concept that after 60 numbers in a fibonacci numbers the last digit repeats itself again and again
+    for(i = 2 ; i < 60 ; i++ ){        //for storing the last digits of squares of first 60 fibonacci numbers
+        a[i] = (a[i-1] + a[i-2]) % 10;       
+        c = a[i];                        
+        b[i] = (c * c) % 10;            //Now we will store last digits in new array and their index as position in fibonacci series
```
Comment (asha15) [inline Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp]: take the comment into previous line
It's better if you make it simple

File: Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    for(i = 2 ; i < 60 ; i++ ){        //for storing the last digits of squares of first 60 fibonacci numbers
+        a[i] = (a[i-1] + a[i-2]) % 10;       
+        c = a[i];                        
+        b[i] = (c * c) % 10;            //Now we will store last digits in new array and their index as position in fibonacci series 
+    } 
+    cin>> n ;  
+
+    cout<< "\nThe Position of Fibonacci Number: " << n ; //getting the required number
+    
+    cout<< "\nThe Last Digit of Square of this Number : " << b[n % 60] ;//print the last digit given in that position
```
Comment (asha15) [inline Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp]: keep space around << n >>

File: Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    //Using the concept that after 60 numbers in a fibonacci numbers the last digit repeats itself again and again
+    for(i = 2 ; i < 60 ; i++ ){        //for storing the last digits of squares of first 60 fibonacci numbers
+        a[i] = (a[i-1] + a[i-2]) % 10;       
+        c = a[i];                        
+        b[i] = (c * c) % 10;            //Now we will store last digits in new array and their index as position in fibonacci series 
+    } 
+    cin>> n ;  
+
+    cout<< "\nThe Position of Fibonacci Number: " << n ; //getting the required number
+
```
Comment (asha15) [inline Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp]: remove this line

File: Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        c = a[i];                        
+        b[i] = (c * c) % 10;            //Now we will store last digits in new array and their index as position in fibonacci series 
+    } 
+    cin>> n ;  
+
+    cout<< "\nThe Position of Fibonacci Number: " << n ; //getting the required number
+    
+    cout<< "\nThe Last Digit of Square of this Number : " << b[n % 60] ;//print the last digit given in that position
+    return 0;
+}
```
Comment (asha15) [inline Square_of_Last_Digit_of_nth_Fibonacci_Number/Square_of_Last_Digit_of_nth_Fibonacci_Number.cpp]: keep a line space before I/O

**Comment by RedEye00** [comment]
plz can clarify my doubts after that i will apply changes

**Comment by asha15** [comment]
@RedEye00 yes,can i know your doubt ?

**Comment by RedEye00** [comment]
I have asked doubts in review section above plz answer

**Comment by asha15** [comment]
could you please tag me nearly the comment ?

**Comment by RedEye00** [comment]
hey @asha15  I have changed all other what you set but i did'nt change the comments in lines 14 and 17 because each comment tells about the specific step so i kept it that way if want me to change plz tell and plz review sorry for delay

**Comment by asha15** [comment]
then take the comment to the previous line ,it looks neat
do these change and let me know

**Comment by RedEye00** [comment]
I have made changes in comments as you said @asha15 plz review

File: Last_Digit_of_Square_of_Nth_Fibonacci_Number/Last_Digit_of_Square_of_Nth_Fibonacci_Number.cpp [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    a[0] = 0;
+    a[1] = 1;
+    b[0] = 0;
+    b[1] = 1;
+    
+    /*Using the concept that after 60 numbers in a fibonacci 
+    numbers the last digit repeats itself again and again*/
+    
+    for(i = 2 ; i < 60 ; i++){        //for storing the last digits of squares of first 60 fibonacci number 
+        a[i] = (a[i - 1] + a[i - 2]) % 10;//Now we will store last digits in new array and their index as position in fibonacci series
```
Comment (asha15) [inline Last_Digit_of_Square_of_Nth_Fibonacci_Number/Last_Digit_of_Square_of_Nth_Fibonacci_Number.cpp]: please take the comments to the previous line

File: Last_Digit_of_Square_of_Nth_Fibonacci_Number/Last_Digit_of_Square_of_Nth_Fibonacci_Number.cpp [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    numbers the last digit repeats itself again and again*/
+    
+    for(i = 2 ; i < 60 ; i++){        //for storing the last digits of squares of first 60 fibonacci number 
+        a[i] = (a[i - 1] + a[i - 2]) % 10;//Now we will store last digits in new array and their index as position in fibonacci series        
+        c = a[i];                        
+        b[i] = (c * c) % 10;            
+    } 
+    cin>> n ;
+    
+    cout<< " \n The Last Digit of Square of this Number : " << b[ n % 60] ;//print the last digit given in that position
```
Comment (asha15) [inline Last_Digit_of_Square_of_Nth_Fibonacci_Number/Last_Digit_of_Square_of_Nth_Fibonacci_Number.cpp]: please take the comments to the previous line

**Comment by RedEye00** [comment]
Sorry @asha15  for delay I have done the changes you requested and thank you for patiently  reviewing the codes

**Comment by asha15** [comment]
Great job

can i know your name?

**Comment by RedEye00** [comment]
Myself A.Rohit Varma

**Pull request merged**